ramls/link.raml

```diff
@@ -0,0 +1,48 @@
+#%RAML 1.0
+title: Note Links
+version: v1.0.0
+baseUri: http://github.com/org/folio/mod-notes
+
+documentation:
+  - title: mod-notes Links API
+    content: This documents the API calls that can be made to query and manage note links of the system
+
+types:
+  noteLinksPut: !include types/link/noteLinksPut.json
+  errors: !include raml-util/schemas/errors.schema
+
+traits:
+  validate: !include raml-util/traits/validation.raml
+  language: !include raml-util/traits/language.raml
+
+/note-links:
+  /domain/{domain}/type/{type}/id/{id}/:
+    put:
```

Comment:
So we have domain, type and id as a means of uniquely defining the link (think we will eventually remove the domain when domain gets refactored to note level).